sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -298,3 +299,36 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 
   override def prettyName: String = "length"
 }
+
+/**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    }
+    else if (string.toString.length == 0) {
+      return UTF8String.fromString(string.toString)
+    }
+    else {
+      val str: String = string.toString
+      val sb = new StringBuilder
```

Comment:
I don't understand your comment. What should I improve? StringBuilder of Scala and Java are more or less identical,
see https://github.com/scala/scala/blob/70f0b1ded880ec9b3a9478d02f1898fcfeee230c/src/library/scala/collection/mutable/StringBuilder.scala

Initialising the `StringBuilder` with the proper size has nothing to do with blank strings or null checks. If you call a StringBuilder without a size, it gets initialised with 16 (see http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/lang/StringBuilder.java#StringBuilder.%3Cinit%3E%28%29). The result is that `StringBuilder` will reserve space for 16 characters. But we know already the size of the result string (`str.length`). This allows us to initialise the `StringBuilder` with the expected size. The advantage is that `ensureCapacity` will never call `expandCapacity`, see http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/lang/AbstractStringBuilder.java#AbstractStringBuilder.ensureCapacity%28int%29 . If the string is not 16 characters long it saves some memory.